src/components/application_manager/rpc_plugins/app_service_rpc_plugin/test/app_service_plugin/app_service_app_extension_test.cc

```diff
@@ -0,0 +1,213 @@
+  ASSERT_EQ(1u, app_service_app_extension_->Subscriptions().size());
+  ASSERT_TRUE(
+      app_service_app_extension_->IsSubscribedToAppService(kAppServiceType1));
+
+  EXPECT_FALSE(
+      app_service_app_extension_->UnsubscribeFromAppService(kAppServiceType2));
+  EXPECT_TRUE(
+      app_service_app_extension_->IsSubscribedToAppService(kAppServiceType1));
+  EXPECT_EQ(1u, app_service_app_extension_->Subscriptions().size());
+}
+
+TEST_F(AppServiceAppExtensionTest,
+       UnsubscribeFromAppService_UnsubscribeAll_SUCCESS) {
+  auto app_service_types = {kAppServiceType1, kAppServiceType2};
+
+  for (const auto& app_service_type : app_service_types) {
+    ASSERT_TRUE(
+        app_service_app_extension_->SubscribeToAppService(app_service_type));
+    ASSERT_TRUE(
+        app_service_app_extension_->IsSubscribedToAppService(app_service_type));
+  }
+  ASSERT_EQ(2u, app_service_app_extension_->Subscriptions().size());
+
+  app_service_app_extension_->UnsubscribeFromAppService();
+
+  for (const auto& app_service_type : app_service_types) {
+    EXPECT_FALSE(
+        app_service_app_extension_->IsSubscribedToAppService(app_service_type));
+  }
+  EXPECT_EQ(0u, app_service_app_extension_->Subscriptions().size());
+}
+
+TEST_F(AppServiceAppExtensionTest, SaveResumptionData_SUCCESS) {
+  ASSERT_TRUE(
+      app_service_app_extension_->SubscribeToAppService(kAppServiceType1));
+  ASSERT_EQ(1u, app_service_app_extension_->Subscriptions().size());
+  ASSERT_TRUE(
+      app_service_app_extension_->IsSubscribedToAppService(kAppServiceType1));
+
+  smart_objects::SmartObject resumption_data;
+  resumption_data[kResumptionDataKey] = "some resumption data";
+
+  app_service_app_extension_->SaveResumptionData(resumption_data);
+
+  EXPECT_TRUE(resumption_data.keyExists(kResumptionDataKey));
+  EXPECT_TRUE(resumption_data.keyExists(kAppServiceInfoKey));
+  EXPECT_EQ(kAppServiceType1,
+            resumption_data[kAppServiceInfoKey][0].asString());
+}
+
+TEST_F(AppServiceAppExtensionTest, ProcessResumption_SUCCESS) {
+  app_service_app_extension_->UnsubscribeFromAppService();
+  ASSERT_EQ(0u, app_service_app_extension_->Subscriptions().size());
+
+  smart_objects::SmartObject app_service_data =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  app_service_data.asArray()->push_back(
+      smart_objects::SmartObject(kAppServiceType1));
+  app_service_data.asArray()->push_back(
+      smart_objects::SmartObject(kAppServiceType2));
+
+  smart_objects::SmartObject resumption_data;
+  resumption_data[kAppServiceInfoKey] = app_service_data;
+
+  app_service_app_extension_->ProcessResumption(resumption_data);
+
+  for (const auto& app_service_type : {kAppServiceType1, kAppServiceType2}) {
+    EXPECT_TRUE(
+        app_service_app_extension_->IsSubscribedToAppService(app_service_type));
+  }
+  EXPECT_EQ(2u, app_service_app_extension_->Subscriptions().size());
+  EXPECT_EQ(kAppServiceType1,
```

Comment:
@YarikMamykin What is the sense of this last check?

Reply:
@AByzhynar fixed in 7fac4d2c0b4cd1d6ce1768602598fa10b9cf6c14